Readability.js

```diff
@@ -1689,6 +1695,26 @@ Readability.prototype = {
   },
 
   /**
+   * Clean out elements whose id/class combinations match specific string.
+   *
+   * @param Element
+   * @param RegExp match id/class combination.
+   * @return void
+   **/
+  _cleanMatchedNodes: function(e, regex) {
+    var next = this._getNextNode(e);
```

Comment:
This is a bit confusing. Earlier you used `_forEachNode` and you already end up calling this for every child of `articleContent`. But `_getNextNode` is used to iterate over the entire document. So now you end up iterating over the entire document once for each child node - duplicating work.

The correct thing to do would likely be to have a `_cleanMatchedNodes` method that takes a regex and an element, and loops over that node and everything after it in the DOM tree until it moves out of the subtree. Note that the code you've written here also skips the element you're being passed, because you call `_getNextNode` immediately. How about this:

```js
_cleanMatchedNodes: function(root, regex) {
  // The second argument means _getNextNode will return the first node it'd iterate
  // over if it wasn't also iterating over itself and child nodes.
  var endOfSearchMarkerNode = this._getNextNode(root, true);
  // Ignore the articleContent node itself when iterating (ie only remove children)
  var n = this._getNextNode(root);
  while (n && n != endOfSearchMarkerNode) {
    if (regex.test(n.className + " " + n.id)) {
      n = this._removeAndGetNext(n);
    } else {
      n = this._getNextNode(n);
    }
  }
}
```